files/usr/share/jeos-firstboot/modules/ssh_enroll

```diff
@@ -0,0 +1,131 @@
+# SPDX-License-Identifier: MIT
+# SPDX-FileCopyrightText: Copyright 2024 SUSE LLC
+# shellcheck shell=bash
+
+ssh_enroll_handle_pubkeys()
+{
+	filename="$1"
+	if ! [ -s "$filename" ]; then
+		d --msgbox $"No public keys received!" 0 0
+		return 0
+	fi
+
+	items=()
+	lineno=1
+	state=on
+	while read -r type key comment; do
+		items+=("$lineno" "${type} ${key:0:8}...${key: -20:20}" "$state")
+		lineno=$((lineno+1))
+		state=off # Only import the first key by default
+	done <"$filename"
+
+	text="$(printf $"Received %d public keys from %s. Select the key(s) to import." "$(wc -l <"$filename")" "$(awk '{print $3; exit}' "$filename")")"
+	d --title $"SSH Key Enrollment" --separate-output --no-tags --checklist "$text" 12 60 10 "${items[@]}"
+
+	readarray -t input <<<"$result"
+	# readarray converts an empty string to ("")
+	if [ -z "${input[0]}" ]; then
+		return 0
+	fi
+
+	(umask 077; mkdir -p ~/.ssh)
+	for lineno in "${input[@]}"; do
+		awk "NR==${lineno}" <"$filename" >> ~/.ssh/authorized_keys
+	done
+
+	text="$(printf $"Imported %d keys" "${#input[@]}")"
+	d --title $"SSH Key Enrollment" --msgbox "$text" 0 0
+}
+
+ssh_enroll_do_config()
+{
+	if ! dialog --backtitle "$PRETTY_NAME" --title $"SSH Key Enrollment" --yesno $"Do you want to enroll keys for SSH access?" 0 0; then
+		return 0
+	fi
+
+	# Stop sshd.service temporarily if needed
+	start_sshd_again=0
+	if systemctl -q is-active sshd.service; then
+		systemctl stop sshd.service
+		start_sshd_again=1
+	fi
+
+	# Make sure host keys exist
+	sshd-gen-keys-start &>/dev/null
+
+	tmpdir=$(umask 077; mktemp -d)
+
+	# Start the server in the background and show a dialog with information while it's running
+	# and offer the option to cancel the process. Unfortunately bash doesn't have a way to wait for
+	# one of multiple processes to exit and get information about which process exited and how.
+	# To work around that, use subprocesses with FIFOs: One to signal once they're up and can handle
+	# SIGTERM properly and one to signal that it exited and how.
+	# See also the welcome-screen implementation.
+	mkfifo "${tmpdir}/serverfifo"
+	mkfifo "${tmpdir}/dialogfifo"
+	mkfifo "${tmpdir}/exitfifo"
+
+	(
+		ssh-pairing-server >"${tmpdir}/pubkeys" 2>"${tmpdir}/serverror" &
+		pid=$!; trap 'kill $pid' SIGTERM; echo server >"${tmpdir}/serverfifo"; wait $pid; echo server $? >>"${tmpdir}/exitfifo"
+	) &
+	serverpid=$!
+
+	{
+		echo $"Please connect to one of the addresses with ssh:"
+		hostname -I
+		echo
+		echo $"Please verify that the host key matches:"
+		for i in /etc/ssh/ssh_host_*key.pub; do ssh-keygen -l -f "$i"; done | awk '{ print $2" "$4 }'
```

Comment:
Maybe you can use `ssh-keygen -lv` there in some way to also display the **visual host key** ASCIII art. I find this a lot easier to compare. You can also add a hint how to invoke `ssh` to display the visual host key upon connect (`ssh -o "VisualHostKey yes"`).

Reply:
I thought about that as well but displaying the random art for all key types and also the ssh client command line in addition to the current dialog content is a bit much, especially if an 80x25 terminal is used.

Reply:
Maybe you can display it at least optionally, if the terminal is large enough. It would be a major usability feature IMHO. It could also be a UI element, like a button, that leaves TUI mode and displays the random art on the text console.

Reply:
Done, see the PR comment